src/platform/intel/cavs/lib/clk.c

```diff
@@ -293,18 +293,29 @@ static int clock_platform_set_cpu_freq(int clock, int freq_idx)
 
 void platform_clock_init(struct sof *sof)
 {
+	uint32_t platform_waiti_clock = 1;
 	int i;
 
 	sof->clocks =
 		cache_to_uncache((struct clock_info *)platform_clocks_info);
 
+#if CAVS_VERSION == CAVS_VERSION_2_5
+	/* Check HW version clock capabilities */
+
+	shim_write(SHIM_CLKCTL, shim_read(SHIM_CLKCTL) | SHIM_CLKCTL_WOV_CRO_REQUEST);
```

Comment:
A small comment is needed explaining the flow and how it determines the clock to be used.

Reply:
added description